benchmark/SoftMaxCrossEntropyBenchmark.jl

```diff
@@ -0,0 +1,63 @@
+include("../src/layers/LayerBase.jl")
+include("../src/layers/SoftMaxCrossEntropy.jl")
+include("../src/layers/SoftMax.jl")
+include("../src/layers/CrossEntropyLoss.jl")
+using Base.Test
+
+smaxcross = SoftMaxCrossEntropyLoss()
+smax = SoftMax()
+cross = CrossEntropyLoss()
+
+
+function before(smaxcross, smax, cross, dict)
+    init(smaxcross, nothing, dict)
+    init(smax     , nothing, dict)
+    init(cross    , nothing, dict)
+end
+
+function benchmmark(smaxcross, smax, cross, batch_size, input_size; alpha = 1.)
+  # batch_size = 10
+  # input_size = 5
+  before(smaxcross, smax, cross, Dict{String, Any}("batch_size" => batch_size, "input_size" => [input_size]))
+  x = rand(batch_size, input_size)
+  label = rand(0:input_size-1, batch_size,1)
+  # benchmark for original SoftMaxCrossEntropy
+  tic()
```

Comment:
I will stick with using`tic() toq()` here since according to documentation  https://docs.julialang.org/en/release-0.4/manual/performance-tips/, "... This is the single biggest advantage of @time vs. functions like tic() and toc(), which only report time"; these two functions are virtually the same but I'll stick with `tic()` here since I would only want time to be reported here since this is a benchmark test in which I want merely a broad sense of time spent by each function; I usually use @time when inspecting into details of methods to figure out whether it has given too many allocations.